src/components/code/index.tsx

```diff
@@ -14,41 +15,202 @@ import styles, {
   content: string;
   language?: string;
-  element?: React.ReactElement;
-  isColorSyntax?: boolean;
   copyButton?: boolean;
   downloadButton?: boolean;
   downloadCallback?: () => void;
   wrapLongLines?: boolean;
   showLineNumbers?: boolean;
   copyCallback?: () => Promise<boolean>;
-  padding?: string;
+  expandable?: boolean;
+  popupProps?: Omit<PopupProps, 'children' | 'isOpen' | 'onClose'>;
 }
 
+export const defaultPopupProps = {
+  width: 'calc(100vw - 80px)',
+  height: 'calc(100vh - 80px)',
+};
+
 const Code: FC<CodeProps> = ({
   title,
   content,
-  language,
-  isColorSyntax,
+  language = 'text',
   copyButton = false,
   downloadButton = false,
-  downloadCallback = undefined,
+  downloadCallback,
   wrapLongLines,
   showLineNumbers,
-  copyCallback = undefined,
-  element,
-  padding = '20px',
+  copyCallback,
+  expandable = false,
+  popupProps = defaultPopupProps,
   ...props
 }: CodeProps) => {
-  const [copied, setCopied] = useState(false);
+  const contentLines = content.split('\n');
 
+  // if it's expandable render it inside ExpandViewer
+  if (expandable && contentLines.length > CONTENT_UPPER_BOUND)
+    return (
+      <ExpandViewer
+        title="Code snippet expanded view"
+        /** currently merging with built-in javascript spread, if we need deep merging we may use lodash merge function instead */
+        popupProps={{ ...defaultPopupProps, ...popupProps }}
+        NormalComponent={() => (
+          <CodeSnippet
+            content={content}
+            contentToShow={content}
+            title={title}
+            language={language}
+            copyButton={copyButton}
+            downloadButton={downloadButton}
+            downloadCallback={downloadCallback}
+            wrapLongLines={wrapLongLines}
+            maxHeightOfCode={
+              popupProps.height || popupProps.maxHeight
+                ? `calc(${popupProps.height ?? popupProps.maxHeight} - 143px)`
+                : '100%'
+            }
+            {...props}
+          />
+        )}
+        BriefComponent={() => (
+          <CodeSnippet
+            content={content}
+            contentToShow={contentLines
+              .slice(0, CONTENT_UPPER_BOUND)
+              .join('\n')}
+            title={title}
+            language={language}
+            copyButton={copyButton}
+            downloadButton={downloadButton}
+            downloadCallback={downloadCallback}
+            wrapLongLines={wrapLongLines}
+            {...props}
+          />
+        )}
+      />
+    );
+
+  // if it's not expandable then render it Normaly
+  return (
+    <CodeSnippet
+      content={content}
+      title={title}
+      language={language}
+      copyButton={copyButton}
+      downloadButton={downloadButton}
+      downloadCallback={downloadCallback}
+      wrapLongLines={wrapLongLines}
+      {...props}
+    />
+  );
+};
+
+export default Code;
+
+// Normal Code Component
+interface CodeSnippetProps
```

Comment:
is it a pattern you like with specifying default export and then below writing other components that are part of the main one?
it looks neat, but I'm not 100% sure

Reply:
Yes, it's the kinda a practice I like. if the components and the main default exported component is not big I would keep them in the same file. 
It depends on how you guys are comfortable with it, if not I can separate them to their own file?